packages/codegen-ui-react/lib/react-component-render-helper.ts

```diff
@@ -127,15 +127,100 @@ export function isActionEvent(event: StudioComponentEvent): event is ActionStudi
   return typeof event === 'object' && 'action' in event;
 }
 
+export function sanitizeString(str: string) {
+  if (typeof str !== 'string') {
+    return '';
+  }
+
+  // Check for dangerous JavaScript patterns
+  const dangerousPatterns = [
+    // JavaScript functions
+    /eval\s*\(/i,
+    /Function\s*\(/i,
+    /setTimeout\s*\(/i,
+    /setInterval\s*\(/i,
+    /new\s+Function/i,
+    /import\s*\(/i,
+    /require\s*\(/i,
+
+    // DOM manipulation
+    /document\./i,
+    /window\./i,
+    /location\./i,
+    /localStorage\./i,
+    /sessionStorage\./i,
+
+    // Event handlers
+    /on\w+\s*=/i, // matches onerror=, onload=, etc.
+
+    // Dangerous protocols
+    /javascript:/i,
+    /data:/i,
+    /vbscript:/i,
+
+    // Script tags and variations
+    /<script/i,
+    /<\/script/i,
+    /<x:script/i,
+
+    // SVG exploits
+    /<svg/i,
+    /xlink:href/i,
+
+    // Object prototype attacks
+    /\[\s*Symbol\s*\./i,
+    /__proto__/i,
+    /prototype\s*\./i,
+
+    // Template literal attacks
+    /\$\{/i,
+
+    // Base64 indicators
+    /base64/i,
+  ];
+
+  if (dangerousPatterns.some((pattern) => pattern.test(str))) {
+    return '';
+  }
+
+  return str.trim();
+}
+
+export function escapePropertyName(propertyName: string): string {
+  // First check if it's a reserved keyword
+  if (keywords.has(propertyName)) {
+    return `${propertyName}Prop`;
+  }
+
+  // Then sanitize the name to ensure it's a valid JavaScript identifier
+  return sanitizeString(propertyName);
```

Comment:
Similar comment to below, are we sure string is the right type here, or are objects/nested objects supported